lutris/util/desktopapps.py

```diff
@@ -0,0 +1,127 @@
+"""Get games installed as desktop applications."""
+
+import os
+import shutil
+import shlex
+import subprocess
+import re
+
+from gi.repository import Gio
+from lutris import pga
+from lutris.util.log import logger
+from lutris.util.strings import slugify
+from lutris.config import make_game_config_id, LutrisConfig
+
+IGNORED_APPS = (
+    "lutris", "mame", "dosbox", "playonlinux", "org.gnome.Games", "retroarch",
+    "steam", "steam-runtime", "steam-valve", "steam-native"
+)
+IGNORED_EXECUTABLES = (
+    "lutris", "steam"
+)
+
+def mark_as_installed(appid, runner_name, game_info):
+    for key in ['name', 'slug']:
+        assert game_info[key]
+    logger.info("Setting %s as installed" % game_info['name'])
+    config_id = (game_info.get('config_path') or make_game_config_id(game_info['slug']))
+    game_id = pga.add_or_update(
+        name=game_info['name'],
+        runner=runner_name,
+        slug=game_info['slug'],
+        installed=1,
+        configpath=config_id,
+        installer_slug=game_info['installer_slug']
+    )
+
+    game_config = LutrisConfig(
+        runner_slug=runner_name,
+        game_config_id=config_id,
+    )
+    game_config.raw_game_config.update({'appid': appid, 'exe': game_info['exe'], 'args': game_info['args']})
+    game_config.save()
+    return game_id
+
+def mark_as_uninstalled(game_info):
+    assert 'id' in game_info
+    assert 'name' in game_info
+    logger.info('Setting %s as uninstalled' % game_info['name'])
+    game_id = pga.add_or_update(
+        id=game_info['id'],
+        runner='',
+        installed=0
+    )
+    return game_id
+
+def sync_with_lutris():
+    apps = get_games()
+    desktop_games_in_lutris = pga.get_desktop_games()
+    slugs_in_lutris = set([str(game['slug']) for game in desktop_games_in_lutris])
+
+    seen_slugs = set()
+    for app in apps:
+        game_info = None
+        name = app[0]
+        slug = slugify(name)
```

Comment:
I will not be surprised when some games exist in Lutris and have a different slug than what this will generate, but not sure there's a good way around that without either 1) lobby for an extension to .desktop for games to use a globally unique identifier (say, an UUID like [MusicBrainz](https://musicbrainz.org/doc/MusicBrainz_Identifier) (which reminds me I still need to make that forum post)), or 2) maintain some sort of alias list in Lutris. (The latter could also be helpful if we need/want to "rename" games in Lutris but avoid breaking old links/URLs/slug references.)

Not really a problem with the PR per se, more an underlying issue that again should maybe be considered. :)